wm.c

```diff
@@ -56,27 +66,43 @@ int main() {
 	hookHandle = SetWindowsHookExW(WH_SHELL, (HOOKPROC)shellProc, wmDll, 0);
 
 	if (hookHandle == NULL) {
-		reportWin32Error(L"SetWindowsHookExW");
+		reportWin32Error(L"SetWindowsHookExW failed for shell hook");
 		goto cleanup;
 	}
-
+	
 	signal(SIGINT, ctrlc);
 
+	//----------------------------------------------
+	/** 
+	 * Handle a message loop
+	**/
+	//----------------------------------------------
 	tileWindows();
+	MSG msg; 
+	while (GetMessage(&msg, NULL, 0, 0) != 0) {
+		//TODO Need to modify this wait so the program can handle hotkeys
```

Comment:
If you go with installing a keyboard hook similar to how the shell hook is installed, you can use WaitForMultipleObjects for example and work with an event for that as well, like how the window event works.